python/Ganga/Core/GangaRepository/GangaRepositoryXML.py

```diff
@@ -1140,9 +1128,9 @@ def delete(self, ids):
                 logger.debug("Delete Error: %s" % err)
             self._internal_del__(this_id)
             rmrf(os.path.dirname(fn))
-            if this_id in self._fully_loaded:
-                del self._fully_loaded[this_id]
             if this_id in self.objects:
+                if self.objects[this_id]._inMemory:
```

Comment:
Since the delete happens anyway, do we really need the `if` here?
Would simply setting 
```
self.objects[this_id]._inMemory = False
```
without the preceding if condition do the same?